### PR TITLE
Add .clang-format for automatic formatting

### PR DESCRIPTION
**Pull request by Xaymar**
This adds clang-format support for automatic formatting based on the Linux Kernel Coding Style. Unfortunately not all options seem possible to change to the correct value, but it's pretty close so far.

### CODE REVIEW

**Comment by Xaymar** [comment]
### Pointer Alignment
Currently this does not align the * of pointer types correctly with the name, it just pushes it one over. This is a result of aligning declaration and assignments, which was present in a large number of libobs file.
```
// old
struct enum_cb {
	obs_enum_audio_device_cb cb;
	void *data;
	int cont;
};
// new
struct enum_cb {
	obs_enum_audio_device_cb cb;
	void *                   data;
	int                      cont;
};
```
The pointer alignment issue also appears with casts, which should not be affected by it at all.

### Defines are not aligned
There seems to be no option to align defines with one another, so the following happens:
```
// old
#define CALL_PARAM_IN  (1<<0)
#define CALL_PARAM_OUT (1<<1)
// new
#define CALL_PARAM_IN (1 << 0)
#define CALL_PARAM_OUT (1 << 1)
```

**Comment by Xaymar** [comment]
Fixed the compiling problems by adjusting the sorting for C and Intrinsic headers to be higher.

File: libobs/audio-monitoring/pulse/pulseaudio-wrapper.h [hunk 1]
```diff
@@ -29,8 +29,8 @@ struct pulseaudio_default_output {
 
 struct enum_cb {
 	obs_enum_audio_device_cb cb;
-	void *data;
-	int cont;
+	void *                   data;
```
Comment (Xaymar) [inline libobs/audio-monitoring/pulse/pulseaudio-wrapper.h:32]: Pointer does not align with name.

File: libobs/audio-monitoring/pulse/pulseaudio-wrapper.h [hunk 2]
```diff
@@ -106,8 +106,8 @@ void pulseaudio_accept();
  *
  * @warning call without active locks
  */
-int_fast32_t pulseaudio_get_source_info_list(pa_source_info_cb_t cb,
-		void *userdata);
+int_fast32_t pulseaudio_get_source_info_list(
+		pa_source_info_cb_t cb, void *userdata);
```
Comment (Xaymar) [inline libobs/audio-monitoring/pulse/pulseaudio-wrapper.h:110]: Wrapped line but first parameter still fits onto the same line.

File: libobs/callback/calldata.h [hunk 3]
```diff
@@ -40,14 +40,14 @@ enum call_param_type {
 	CALL_PARAM_TYPE_STRING
 };
 
-#define CALL_PARAM_IN  (1<<0)
-#define CALL_PARAM_OUT (1<<1)
+#define CALL_PARAM_IN (1 << 0)
+#define CALL_PARAM_OUT (1 << 1)
```
Comment (Xaymar) [inline libobs/callback/calldata.h:44]: Defines no longer aligned with each other.

File: libobs/graphics/bounds.h [hunk 4]
```diff
@@ -82,39 +82,30 @@ EXPORT bool bounds_intersection_line(const struct bounds *b,
...
 {
-	return test->min.x >= b->min.x &&
-	       test->min.y >= b->min.y &&
-	       test->min.z >= b->min.z &&
-	       test->max.x <= b->max.x &&
-	       test->max.y <= b->max.y &&
-	       test->max.z <= b->max.z;
+	return test->min.x >= b->min.x && test->min.y >= b->min.y
+			&& test->min.z >= b->min.z && test->max.x <= b->max.x
+			&& test->max.y <= b->max.y && test->max.z <= b->max.z;
```
Comment (Xaymar) [inline libobs/graphics/bounds.h:91]: Readable test converted into unreadable mess.

File: libobs/graphics/device-exports.h [hunk 5]
```diff
@@ -45,68 +45,57 @@ EXPORT gs_texture_t *device_texture_create(gs_device_t *device, uint32_t width,
...
+		const char *shader, const char *file, char **error_string);
+EXPORT gs_vertbuffer_t *device_vertexbuffer_create(
+		gs_device_t *device, struct gs_vb_data *data, uint32_t flags);
+EXPORT gs_indexbuffer_t *   device_indexbuffer_create(gs_device_t *device,
+		   enum gs_index_type type, void *indices, size_t num,
+		   uint32_t flags);
+EXPORT enum gs_texture_type device_get_texture_type(const gs_texture_t *texture);
+EXPORT void                 device_load_vertexbuffer(
+				gs_device_t *device, gs_vertbuffer_t *vertbuffer);
+EXPORT void device_load_indexbuffer(
```
Comment (Xaymar) [inline libobs/graphics/device-exports.h:70]: Alignment lost because of incorrect wrapping.

File: libobs/graphics/effect-parser.h [hunk 6]
```diff
@@ -47,7 +47,7 @@ enum ep_var_type {
 };
 
 struct ep_var {
-	char *type, *name, *mapping;
+	char *           type, *name, *mapping;
 	enum ep_var_type var_type;
```
Comment (Xaymar) [inline libobs/graphics/effect-parser.h:51]: Multiple variables on a single line, should be fixed in origin source.

File: libobs/graphics/input.h [hunk 7]
```diff
@@ -23,123 +23,123 @@
...
+#define KBC_HOME 0x5B
+#define KBC_UP 0x5C
+#define KBC_PAGEDOWN 0x5D
+#define KBC_LEFT 0x5E
+#define KBC_RIGHT 0x5F
+#define KBC_END 0x60
+#define KBC_DOWN 0x61
+#define KBC_PAGEUP 0x62
+#define KBC_INSERT 0x63
+#define KBC_DELETE 0x64
```
Comment (Xaymar) [inline libobs/graphics/input.h:126]: Lost alignment due to lack of define alignement options.

File: libobs/util/serializer.h [hunk 8]
```diff
@@ -151,12 +150,12 @@ static inline void s_wb64(struct serializer *s, uint64_t u64)
 
 static inline void s_wbf(struct serializer *s, float f)
 {
-	s_wb32(s, *(uint32_t*)&f);
+	s_wb32(s, *(uint32_t *)&f);
```
Comment (Xaymar) [inline libobs/util/serializer.h:153]: Additional padding before * in c-style cast.

File: libobs/graphics/device-exports.h (hunk 5 above)
Comment (Xaymar) [reply libobs/graphics/device-exports.h:70]: Probably will disable all alignment options due to this, they don't seem to add to much to the actual readability of the code.

File: libobs/audio-monitoring/pulse/pulseaudio-wrapper.h (hunk 1 above)
Comment (Xaymar) [reply libobs/audio-monitoring/pulse/pulseaudio-wrapper.h:32]: Probably will disable all alignment options due to this, they don't seem to add to much to the actual readability of the code.

**Comment by jp9000** [comment]
I'm going to handle this personally a bit later, probably some time in April.  The reason is because I want to get a whole bunch of pull requests implemented, then I'm going to probably run clang-format on the project (finally).  The reason why this is delicate is because the moment we run it, it will invalidate just about every pull request we have, and currently there are a bit too many.  There are issues that aren't really solvable with .clang-format at the moment, so once it's changed we'll have to live with the changes it makes.

In other words let me take the reigns on this one when I'm ready and the pull request count is down in April.

**Comment by Xaymar** [comment]
Why did you close a draft PR? It was not ready for review.
